src/stringify.js

```diff
@@ -124,7 +123,35 @@ var BASE_16 = 16;
         var mind = gap;
         var partial;
         var value = holder[key];
-        var isInt64 = value !== null && value instanceof Int64;
+        var isInt64 = true;
+        var Int64Keys = ["buffer", "offset"];
+        var Int64Types = [Uint8Array, Number];
+        var valueKeys = [];
+
+        if (value !== null && value !== undefined)
+        {
+            valueKeys = Object.keys(value).sort();
+            // Since javascript is duck-typed, we check the keys of the object and their types. Built-in types are ok.
+            if (valueKeys.length === Int64Keys.length)
+            {
+                for (var it = 0; isInt64 && it < valueKeys.length; ++it)
+                {
+                    if ((valueKeys[it] !== Int64Keys[it]) || (!value[valueKeys[it]] instanceof Int64Types[it]))
+                    {
+                        isInt64 = false;
+                        break;
+                    }
+                }
+            }
+            else
+            {
+                isInt64 = false;
+            }
+        }
+        else
+        {
+            isInt64 = false;
+        }
 
```

Comment:
if it is an int64 value, shouldn't we return right away (with the proper value), so we don't do unnecessary work?

Reply:
actually, in general, we seem to be checking things like is the value an 'object' multiple times, isn't it possible to reorg the code to clean this up a bit (i.e., avoid doing the same check as well as unnecessary work multiple times)?